common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -80,6 +81,7 @@ public void run() {
                     SingleThreadEventExecutor.class, ThreadProperties.class, "threadProperties");
 
     private final Queue<Runnable> taskQueue;
+    private AtomicInteger pendingTaskSize = new AtomicInteger(0);
```

Comment:
I think all these changes should go into `EpollEventLoop` / `NioEventLoop` / `KQueueEventLoop` and the `SingleThreadEventExecutor` should not be changed. This is mainly because otherwise it may produce unexpected effects for people that extend the `SingleThreadEventExecutor`.

Reply:
I'm not sure how I can track how many events were run with `runAllTasks(long timeoutNanos)` then?